### DISCUSSION TITLE
run loader again to get latest data

### DISCUSSION THREAD

**Discussion by ghost** [remix-run/remix#2699] [Q&A] [answered]
Is it possible to call loader from the client side, like(refresh session, refresh the list of data from database)?

**Comment by kiliman** [comment, marked as answer]
Read this article for details
https://sergiodxa.com/articles/automatic-revalidation-in-remix

I also believe the `useDataRefresh` hook from `remix-utils` is the latest iteration of the concept.

https://github.com/sergiodxa/remix-utils#usedatarefresh

**Reply by ghost** [reply]
when i try to use the refresh exported from useDataRefresh, it throw an error: `Uncaught (in promise) Error: No matches found`

**Reply by sergiodxa** [reply]
You need to create the /dev/null route with an action returning null

**Reply by ghost** [reply]
thank you so much, the problem now solved and it can refresh correctly
